### PR TITLE
prevent infinite loops

### PR DESCRIPTION
**Pull request by andriytyurnikov**
via prevent_depth option for backwards compatibility

### CODE REVIEW

**Comment by andriytyurnikov** [comment]
@jeremyevans do you think some other function definitions could use same option as well?

File: lib/sequel/extensions/pg_triggers.rb [hunk 1]
```diff
@@ -329,6 +331,14 @@ def pgt_trigger(table, trigger_name, function_name, events, definition, opts={})
       def pgt_mangled_table_name(table)
         quote_schema_table(table).gsub('"', '').gsub(/[^A-Za-z0-9]/, '_').gsub(/_+/, '_')
       end
+
+      def pgt_pg_trigger_depth_guard_clause
+        <<-SQL
+        IF pg_trigger_depth() <> 1 THEN
```
Comment (jeremyevans) [inline lib/sequel/extensions/pg_triggers.rb:337]: This doesn't use the `:prevent_depth` option.  Something like ` pg_trigger_depth() > #{opts[:prevent_depth]}`, pass the `opts` to `pgt_pg_trigger_depth_guard_clause`. and add the conditional inside `pgt_pg_trigger_depth_guard_clause` so you don't need it in all callers.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I think it would be a good idea to support this for all triggers that could results in additional triggers.

File: lib/sequel/extensions/pg_triggers.rb (hunk 1 above)
Comment (andriytyurnikov) [reply lib/sequel/extensions/pg_triggers.rb]: Done

**Comment by jeremyevans** [comment]
Looks good.   I'll try to test and merge this today.

**Comment by andriytyurnikov** [comment]
I am not sure I have context for such generalization, so I only altered definitions for  counter, sum, and many to many sum.

Also, I am somewhat concerned about depth as a numerical parameter in case of sum and counters, and maybe `track_recursive_trigger` is a better option or worthy alternative - but this is beyond my depth of understanding at the moment.

**Comment by andriytyurnikov** [comment]
After giving it some thought - I've added guard clause to every other function definition (including to previously missing for many to many summation). It's relevance might vary on case by case basis, but having access via simple parameter seems like a good thing.

**Comment by jeremyevans** [comment]
I squash-merged your first two commits as f32ebdd5aa83373548fa34d1f982b4ec1135c2d4 (I started working on that before your third and fourth commits).  I also renamed the option to :trigger_depth_limit and added some minor changes and more tests in d68b63102759bd5348c934e88d987809d6e6777e.

I added support for this to some methods, but not all.  I only added it to methods that define trigger functions that result in additional triggers.  I don't see a reason to support the option if the trigger function defined cannot result in additional triggers.

Please review and let me know if you think we should make further changes.

**Pull request closed without merging**